src/db.ts

```diff
@@ -39,12 +40,24 @@ export class DB {
 
 	public createOrUpdateCollection<T extends TigrisCollectionType>(
 		collectionName: string, schema: TigrisSchema<T>): Promise<Collection<T>> {
-		return new Promise<Collection<T>>((resolve, reject) => {
-			const rawJSONSchema: string = Utility._toJSONSchema(collectionName, schema);
+		return this.createOrUpdate(collectionName, schema,
+			() => new Collection(collectionName, this._db, this.grpcClient));
+	}
+
+	public createOrUpdateTopic<T extends TigrisCollectionType>(
```

Comment:
should we share `TigrisCollectionType` with Topic or create another type `TigrisTopicType`?